src/Data/Aeson/Parser/Internal.hs

```diff
@@ -4,7 +4,7 @@
 {-# LANGUAGE OverloadedStrings #-}
 #if __GLASGOW_HASKELL__ <= 710 && __GLASGOW_HASKELL__ >= 706
 -- Work around a compiler bug
-{-# OPTIONS_GHC -fsimpl-tick-factor=300 #-}
+{-# OPTIONS_GHC -fsimpl-tick-factor=400 #-}
```

Comment:
It looks like `300` is enough, but we need to include GHC-8.0.2, i.e. change `__GLASGOW_HASKELL__ <= 710` to `__GLASGOW_HASKELL__ <= 800`.